update/genesis/import.go

```diff
@@ -80,47 +79,54 @@ func NewStateImport(args ArgsNewStateImport) (*stateImport, error) {
 		trieStorageManagers: args.TrieStorageManagers,
 		storageConfig:       args.StorageConfig,
 		shardID:             args.ShardID,
+		hardforkStorer:      args.HardforkStorer,
 	}
 
 	return st, nil
 }
 
 // ImportAll imports all the relevant files for the new genesis
 func (si *stateImport) ImportAll() error {
-	files := si.reader.GetFileNames()
-	if len(files) == 0 {
-		return update.ErrNoFileToImport
-	}
-
-	var err error
-	for _, fileName := range files {
-		switch fileName {
-		case MetaBlockFileName:
-			err = si.importMetaBlock()
-		case MiniBlocksFileName:
-			err = si.importMiniBlocks()
-		case TransactionsFileName:
-			err = si.importTransactions()
+	var errFound error
+
+	si.hardforkStorer.RangeKeys(func(identifier string, keys [][]byte) bool {
```

Comment:
previously this created a channel with a goroutine. Is it ok if it runs outside of a goroutine?

Reply:
yes, in order to have a feedback that the process has ended. Otherwise we need a sort of channel/callback function in order to signal the end of calls which will defeat the purpose of this impl.

Reply:
ok!